app/src/main/java/org/schabi/newpipe/player/mediaitem/MediaItemTag.java

```diff
@@ -61,12 +61,11 @@ default Optional<Quality> getMaybeQuality() {
 
     @NonNull
     static Optional<MediaItemTag> from(@Nullable final MediaItem mediaItem) {
-        if (mediaItem == null || mediaItem.localConfiguration == null
-                || !(mediaItem.localConfiguration.tag instanceof MediaItemTag)) {
-            return Optional.empty();
-        }
-
-        return Optional.of((MediaItemTag) mediaItem.localConfiguration.tag);
+        return Optional.ofNullable(mediaItem)
+                .map(item -> item.localConfiguration)
+                .map(localConfiguration -> localConfiguration.tag)
```

Comment:
`map` will crash if the return value of the lambda is `null`. I looked it up but there does not seem to be a `mapNullable` (just like `ofNullable` exists). So I used `flatMap(... -> Optional.ofNullable(...))`

```java
jshell> Optional.of(new Object()).map(null)
|  Exception java.lang.NullPointerException
|        at Objects.requireNonNull (Objects.java:233)
|        at Optional.map (Optional.java:256)
|        at (#2:1)

jshell> Optional.of(new Object()).flatMap(o -> Optional.ofNullable(null))
$4 ==> Optional.empty
```

Reply:
That's weird, `null` values should be converted to empty `Optional`s: https://developer.android.com/reference/java/util/Optional#map(java.util.function.Function%3C?%20super%20T,%20?%20extends%20U%3E)

Reply:
I will revert that part of my commit

Reply:
I pushed ad605e2c5a38e57af1b0ec7d4c991a9a9240dfff